zerver/lib/bugdown/__init__.py

```diff
@@ -1576,21 +1607,25 @@ def prepare_realm_pattern(source: str) -> str:
 
 # Given a regular expression pattern, linkifies groups that match it
 # using the provided format string to construct the URL.
-class RealmFilterPattern(markdown.inlinepatterns.Pattern):
+class RealmFilterProcessor(markdown.inlinepatterns.InlineProcessor):
     """ Applied a given realm filter to the input """
 
     def __init__(self, source_pattern: str,
                  format_string: str,
                  markdown_instance: Optional[markdown.Markdown]=None) -> None:
         self.pattern = prepare_realm_pattern(source_pattern)
         self.format_string = format_string
-        markdown.inlinepatterns.Pattern.__init__(self, self.pattern, markdown_instance)
+        super().__init__(self.pattern, markdown_instance)
 
-    def handleMatch(self, m: Match[str]) -> Union[Element, str]:
+    def handleMatch(self, m: Match[str], data: str) -> Tuple[Optional[Union[Element, str]],
+                                                             Optional[int], Optional[int]]:
         db_data = self.markdown.zulip_db_data
-        return url_to_a(db_data,
-                        self.format_string % m.groupdict(),
-                        m.group(OUTER_CAPTURE_GROUP))
+        substituted_url = self.format_string % m.groupdict()
+        url_element = url_to_a(db_data, substituted_url, m.group(OUTER_CAPTURE_GROUP))
+        if isinstance(url_element, str):
+            return None, None, None
```

Comment:
This probably deserves a comment explaining why the result being a `str` means that it's a failed match.